mozetl/taar/taar_locale.py

```diff
@@ -37,6 +40,29 @@ def store_new_state(source_file_name, s3_dest_file_name, s3_prefix, bucket):
     transfer.upload_file(source_file_name, bucket, key_path)
 
 
+def load_amo_external_whitelist():
+    final_whitelist = []
+    amo_dump = {}
+    try:
+        # Load the most current AMO dump JSON resource.
+        s3 = boto3.client('s3')
+        s3_contents = s3.get_object(Bucket=AMO_DUMP_BUCKET, Key=AMO_DUMP_KEY)
+        amo_dump = json.loads(s3_contents['Body'].read())
+    except ClientError:
+        logger.exception("Failed to download from S3", extra={
+            "bucket": AMO_DUMP_BUCKET,
+            "key": AMO_DUMP_KEY})
+
+    # If the load fails, we will have an empty whitelist, this may be problematic.
```

Comment:
If the whitelist fails to load, should the job fail or continue?

Reply:
There's an assert on line 193: do you want me to do something different/move it?

Reply:
Can you raise an [appropriate exception](https://docs.python.org/2/library/exceptions.html) instead of the assertion? It can be here or in 193.